src/test/regress/sql/multi_select_distinct.sql

```diff
@@ -0,0 +1,237 @@
+--
+-- MULTI_SELECT_DISTINCT
+--
+-- Tests select distinct, and select distinct on features.
+--
+
+-- unsupported cases
+-- function calls are not supported
+SELECT DISTINCT l_orderkey, now() FROM lineitem_hash_part;
+
+-- const expressions are supported
+SELECT DISTINCT l_orderkey, 1+1 FROM lineitem_hash_part ORDER BY 1 LIMIT 5;
+-- but non consts are not supported
+SELECT DISTINCT l_orderkey, l_partkey + 1 FROM lineitem_hash_part ORDER BY 1 LIMIT 5;
+
+-- column expressions are not supported
+SELECT DISTINCT l_orderkey, l_shipinstruct || l_shipmode FROM lineitem_hash_part;
+
+-- function calls with const input are supported
+SELECT DISTINCT l_orderkey, strpos('AIR', 'A') FROM lineitem_hash_part ORDER BY 1,2 LIMIT 5;
+
+-- function calls with non-const input are supported
+SELECT DISTINCT l_orderkey, strpos(l_shipmode, 'A') FROM lineitem_hash_part;
+
+-- distinct on partition column
+-- verify counts match with respect to count(distinct)
+CREATE TEMP TABLE temp_orderkeys AS SELECT DISTINCT l_orderkey FROM lineitem_hash_part;
+SELECT COUNT(*) FROM temp_orderkeys;
+SELECT COUNT(DISTINCT l_orderkey) FROM lineitem_hash_part;
+
+SELECT DISTINCT l_partkey FROM lineitem_hash_part WHERE l_orderkey < 5 order by 1;
+
+-- distinct on non-partition column
+SELECT DISTINCT l_partkey FROM lineitem_hash_part WHERE l_orderkey > 5 and l_orderkey < 20 order by 1;
+
+SELECT DISTINCT l_shipmode FROM lineitem_hash_part ORDER BY 1 DESC;
+
+-- distinct with multiple columns 
+SELECT DISTINCT l_orderkey, o_orderdate
+	FROM lineitem_hash_part JOIN orders_hash_part ON (l_orderkey = o_orderkey) 
+	WHERE l_orderkey < 10
+	ORDER BY l_orderkey;
+
+-- distinct on partition column with aggregate
+-- this is the same as non-distinct version due to group by
+SELECT DISTINCT l_orderkey, count(*)
+	FROM lineitem_hash_part
+	WHERE l_orderkey < 200
+	GROUP BY 1
+	HAVING count(*) > 5
+	ORDER BY 2 DESC, 1;
+	
+-- explain the query to see actual plan
+EXPLAIN (COSTS FALSE)
+	SELECT DISTINCT l_orderkey, count(*)
+		FROM lineitem_hash_part
+		WHERE l_orderkey < 200
+		GROUP BY 1
+		HAVING count(*) > 5
+		ORDER BY 2 DESC, 1;
+
+-- distinct on non-partition column with aggregate
+-- this is the same as non-distinct version due to group by
+SELECT DISTINCT l_partkey, count(*)
+	FROM lineitem_hash_part
+	GROUP BY 1 
+	HAVING count(*) > 2
+	ORDER BY 1;
+	
+-- explain the query to see actual plan
+EXPLAIN (COSTS FALSE)
+	SELECT DISTINCT l_partkey, count(*)
+		FROM lineitem_hash_part
+		GROUP BY 1 
+		HAVING count(*) > 2
+		ORDER BY 1;
+
+-- following is a bug needs to be fixed
```

Comment:
I found this later in the testing. I need to pay closer look. It should not be difficult. perhaps 0.5 day

Reply:
Could you spend a bit more time and see how complex the fix? The workaround is just to drop distinct as long as I see, but it would be nice to run it as it is. (Also making sure that it doesn't break something else)

Reply:
Yes, dropping distinct resolves it